lux/extensions/auth/models.py

```diff
@@ -29,6 +30,74 @@
 )
 
 
+class Acl(Model):
+    """
+    View-based model
+    """
+    __tablename__ = 'acl'
+    user_id = Column(Integer, primary_key=True)
+    resource = Column(String, primary_key=True)
+    res_id = Column(String, primary_key=True)
+    action = Column(String, primary_key=True)
+    effect = Column(String, primary_key=True)
+
+    def __repr__(self):
+        return "%s:%s:%s:%s:%s" % (
+            self.user_id, self.resource, self.res_id, self.action, self.effect)
+
+    __str__ = __repr__
+
+    __create_sql__ = """
+    create or replace view acl as (
+    with
+        -- QUERY
+        _users_groups as (
+          select
+            u.id as user_id,
+            array_agg(ug.group_id) as group_ids
+          from "user" u
+          inner join users_groups ug on u.id = ug.user_id
+          group by u.id
+        ),
+
+        _users_permissions as (
+          select
+            ug.user_id,
+            array_agg(gp.permission_id) as permission_ids
+          from _users_groups ug
+          inner join groups_permissions gp on gp.group_id = any(ug.group_ids)
+          group by ug.user_id
+        ),
+
+        _single_permissions as (
+          select
+            id,
+            split_part(policy->>'resource', ':', 1) as resource,
+            nullif(split_part(policy->>'resource', ':', 2), '') as res_id,
+            trim(both '"' from jsonb_array_elements(policy->'actions')::text)
+                as "action",
+            coalesce(policy->>'effect', 'allow') as effect
+          from permission p
+        ),
+
+        _permissions_view as (
+          select
+            up.user_id,
+            sp.resource,
+            coalesce(sp.res_id, '*') as res_id,
+            sp.action,
+            sp.effect
+          from _users_permissions up
+          inner join _single_permissions sp on sp.id = any(up.permission_ids)
+          group by up.user_id, up.permission_ids,
+                    sp.resource, sp.res_id, sp.action, sp.effect
+        )
+
```

Comment:
It can be rewritten to select-based format, but I think it's not final version and for now readability may be more important. It's very easy to replace the view in the future.
I'll rewrite it if we decide otherwise